tests/test_alignment.py

```diff
@@ -230,6 +226,78 @@ def test_validate_wcs_list_invalid(wcs_list, expected_error):
     assert type(exec_info.value) == expected_error
 
 
+def get_fake_wcs():
+    fake_wcs1 = fitswcs.WCS(
+        fits.Header(
+            {
+                "NAXIS": 2,
+                "NAXIS1": 4,
+                "NAXIS2": 4,
+                "CTYPE1": "RA---TAN",
+                "CTYPE2": "DEC--TAN",
+                "CRVAL1": 0,
+                "CRVAL2": 0,
+                "CRPIX1": 1,
+                "CRPIX2": 1,
+                "CDELT1": -0.1,
+                "CDELT2": 0.1,
+            }
+        )
+    )
+    fake_wcs2 = fitswcs.WCS(
+        fits.Header(
+            {
+                "NAXIS": 2,
+                "NAXIS1": 5,
+                "NAXIS2": 5,
+                "CTYPE1": "RA---TAN",
+                "CTYPE2": "DEC--TAN",
+                "CRVAL1": 0,
+                "CRVAL2": 0,
+                "CRPIX1": 1,
+                "CRPIX2": 1,
+                "CDELT1": -0.05,
+                "CDELT2": 0.05,
+            }
+        )
+    )
+    return fake_wcs1, fake_wcs2
+
+
+@pytest.mark.parametrize(
+    "x_inp, y_inp, x_expected, y_expected",
+    [
+        (1000, 2000, np.array(2000), np.array(4000)),  # string input test
+        ([1000], [2000], np.array(2000), np.array(4000)),  # array input test
+        pytest.param(1, 2, 3, 4, marks=pytest.mark.xfail),  # expected failure test
+    ],
+)
+def test_reproject(x_inp, y_inp, x_expected, y_expected):
+    wcs1, wcs2 = get_fake_wcs()
+    f = reproject(wcs1, wcs2)
+    x_out, y_out = f(x_inp, y_inp)
+    assert np.allclose(x_out, x_expected, rtol=1e-05)
+    assert np.allclose(y_out, y_expected, rtol=1e-05)
+
+
+def test_wcs_bbox_from_shape_2d():
+    bb = wcs_bbox_from_shape((512, 2048))
+    assert bb == ((-0.5, 2047.5), (-0.5, 511.5))
+
+
+@pytest.mark.parametrize(
+    "shape, pixmap_expected_shape",
+    [
+        (None,(4,4,2)),
+        ((100, 200), (100, 200, 2)),
+    ],
+)
+def test_calc_pixmap(shape, pixmap_expected_shape):
+    wcs1, wcs2 = get_fake_wcs()
```

Comment:
This function is testing only FITS WCS. It'd be good to have a test with gwcs.WCS as well.